runtime/connectors/connectors.go

```diff
@@ -95,6 +97,14 @@ type SamplePolicy struct {
 	Limit    int
 }
 
+// Config is common config for all connectors
+// Different connectors may add more configs
+type Config struct {
+	Path         string `mapstructure:"path"`
+	Format       string `mapstructure:"format"`
+	CSVDelimiter string `mapstructure:"csv.delimiter"`
+}
```

Comment:
Let's keep this decentralized in the separate connectors, even if it means duplication. It may diverge quite soon (we'll probably need SQL and Kafka connectors soon, where these options do not apply)

Reply:
With SQL and Kafka connectors in place, this entire code may need restructuring(Consuming via files may no longer apply).
If not creating these configs only option is to read configs from source.Properties map. In my opinion its never a good idea to read data (like configs) from a map. Makes things easier at start but becomes tough later on to keep track of where and what all data was read.

Reply:
Another thing we can do is let connectors expose fileParsingOptions which we can consume here. Where such options do not apply, connectors will not return valid options.

Reply:
I understand the problem around needing access to this info. I thought some more about this. Some thoughts:

- I agree that reading config from maps in general is not good, though it can be acceptable in some cases if it prevents tight coupling of packages.
- My main worry here is about using struct embedding (inheritance?) for a simple config object, it feels like too much complexity versus simply having a few duplicated lines. The idea of having `FileParseOptions` seems nicer, although maybe premature for so few options.
- About the worry of exposing `Format` and `CSVDelimiter` to the DuckDB driver, I think the solution is to have these be available on the `FileIterator` previously discussed instead. So the DuckDB driver should not need to re-parse the source properties, instead the iterator it gets from the connector should contain any info necessary to correctly consume the files. This also ensures looser coupling.
- For SQL and Kafka, file-based ingestion might still be a nice naive solution. Buffering in files (for X seconds from Kafka, or for X size when loading a SQL query) and then loading the file is a pretty common way to ingest into OLAP DBs – often, `INSERT`s have poor performance for columnar data. (E.g. Druid doesn't even support `INSERT`, the main approaches are load from file in object storage or direct connect to Kafka).
  - But I agree more flexibility might also be needed at some point. One idea is to also offer a `ConsumeAsArrow` option. We will need to scope it when that happens (it will also largely depend on underlying DB capabilities when we get to cloud scale).